### PR TITLE
Fix: Improve feature detection to support Firefox 93

### LINKED ISSUE #45: Latest Firefox Developer Edition/nightlies break polyfill
This is an issue only affecting Firefox Developer Edition and experimental/nightly versions so I don't know if I'd call it a bug, but it tripped me up so I thought it might be useful to others to mention it here.

<details>
<summary>
Today I updated Firefox Developer Edition from 91.0 to 93.0b3 and discovered that form-associated custom elements are no longer being polyfilled.
</summary>

Specifically, after doing `this._internals = this.attachInternals()` I found that `this._internals.setFormValue()` was throwing the error `Uncaught TypeError: this._internals.setFormValue is not a function`. The value of `this._internals` was an EventInternals object, as expected, but it didn't have a `setFormValue` method. It had but a single property, `shadowRoot`.

What's happened is that ElementInternals is _partially_ (but mostly not) implemented in Firefox, so `window.ElementInternals` exists and element-internals-polyfill's check for its absence fails and the polyfill doesn't polyfill. This incomplete implementation lives behind a configuration flag called `dom.webcomponents.elementInternals.enabled`, which is set to `false` in standard Firefox releases but, as of 93.0, set to `true` in Firefox Developer Edition and (I think) experimental/nightly releases.

Firefox [bug 1552313](https://bugzilla.mozilla.org/show_bug.cgi?id=1552313) is related to this, but I'm not sure it's _the_ relevant bug.
</details>

The fix—i.e. to make `window.ElementInternals` `undefined` again so the polyfill does its work—is to go to [`about:config`](https://support.mozilla.org/en-US/kb/about-config-editor-firefox) and search for `dom.webcomponents.elementInternals.enabled` and toggle it to `false`.

### PR DESCRIPTION
**Pull request by lideen**
This is an attempt to solve #45. Needed to fix this for one of my projects, not sure if this is the best way to do this, but I thought I'd share. 😄 

Closes #45

### CODE REVIEW

**Comment by brion-fuller** [comment]
This worked great for me also.

File: src/element-internals.ts [hunk 1]
```diff
@@ -251,7 +251,37 @@ if (!window.CustomStateSet) {
...
+    "states",
+    "willValidate",
+    "validity",
+    "validationMessage",
+    "labels",
+    "setFormValue",
+    "setValidity",
+    "checkValidity",
+    "reportValidity"
+  ].every(prop => prop in new ElementInternalsFeatureDetection().internals)
```
Comment (victorz) [inline src/element-internals.ts:281]: Not sure if it's a concern, but the new object is created here for every element in the property name array.

Comment (lideen) [reply src/element-internals.ts]: Thank you, sorted out in a2a22cd

**Comment by calebdwilliams** [comment]
This is live as part of version `0.1.45`. Thanks @lideen!

**Pull request merged**